src/neo4j_genai/experimental/pipeline/pipeline.py

```diff
@@ -125,35 +118,22 @@ async def execute(self, **kwargs: Any) -> RunResult | None:
         """
         logger.debug(f"Running component {self.name} with {kwargs}")
         start_time = default_timer()
-        try:
-            await self.set_status(RunStatus.RUNNING)
-        except PipelineStatusUpdateError:
-            logger.info(f"Component {self.name} already running or done {self.status}")
-            return None
         component_result = await self.component.run(**kwargs)
-        await self.set_status(RunStatus.DONE)
         run_result = RunResult(
-            status=self.status,
             result=component_result,
         )
         end_time = default_timer()
         logger.debug(f"Component {self.name} finished in {end_time - start_time}s")
         return run_result
 
-    def reinitialize(self) -> None:
-        self.status = RunStatus.SCHEDULED
-
-    async def run(
-        self, inputs: dict[str, Any], callback: PartialOnTaskCompleteProtocol
-    ) -> None:
+    async def run(self, inputs: dict[str, Any]) -> RunResult | None:
         """Main method to execute the task."""
         logger.debug(f"TASK START {self.name=} {inputs=}")
         res = await self.execute(**inputs)
         if res is None:
```

Comment:
In what cases will `res` be `None` again?

Reply:
Right, it doesn't happen anymore. It used to be when the task `run` is called but it was already started in another process. But this check is now done in the orchestrator. So removing this `if`, good catch :)